types/index.d.ts

```diff
@@ -135,13 +135,25 @@ export class Rule implements RuleProperties {
   ): T extends true ? string : RuleSerializable;
 }
 
+export type FactHandler = (
+    result: {
+      fact: string;
+      operator: string;
+      value: { fact: string } | any;
+      receivedValue: { fact: string } | any
+    },
+    almanac: Almanac,
+) => void;
+
```

Comment:
We shouldn't union with any: a union with any with always result in any.
microsoft/TypeScript#18568

To take a guess, the signature should be more like:
```ts
export type FactHandler<T = string> = (
    result: {
      fact: Fact<T>;
      operator: string;
      value: { fact: T }
      receivedValue: { fact: T }
    },
    almanac: Almanac,
) => void

interface ConditionProperties<T = string> {
  fact: string;
  operator: string;
  value: { fact: T };
  path?: string;
  priority?: number;
  params?: Record<string, any>;
  onSuccess?: FactHandler<T>;
  onFailure?: FactHandler<T>;
}
```